.github/workflows/visual-regression.yml

```
@@ -0,0 +1,80 @@
+        path: |
+          test-screenshots/
+        retention-days: 7
+        
+    - name: Upload baseline screenshots (on main branch)
+      uses: actions/upload-artifact@v4
+      if: github.ref == 'refs/heads/main'
+      with:
+        name: visual-regression-baseline
+        path: |
+          test-screenshots/baseline/
+        retention-days: 30
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)